### ISSUE TITLE
sdg_op component fails with RHELAI 1.3

### ISSUE THREAD

**Issue by MichaelClifford** [opendatahub-io/ilab-on-ocp#223]
`sdg_op` with RHEL 1.3 fails at line 76 due to a change in how `read_taxonomy` now works. It now expects an additional parameter `document_output_dir` to be present.   https://github.com/opendatahub-io/ilab-on-ocp/blob/e57915f4f03dbf2f65e33fc9944fb24ac1f28047/sdg/components.py#L76 

This is the error:
```
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

That said, it is unclear to me that line 76 here is doing much. I recommend simply removing it as `read_taxonomy` is called inside of the `generate_data` function anyways.

**Comment by tumido** [comment]
This error is interesting... Since the source code in the open source ILab says this param is fully optional..

https://github.com/instructlab/sdg/blob/eef8baedf36f21da73bda39270ed7378558541ad/src/instructlab/sdg/utils/taxonomy.py#L314-L319

**Comment by MichaelClifford** [comment]
Update: 

Looks like `/usr/share/instructlab/sdg/default_data_recipes/skills.yaml` is no longer editable by the user. This causes the `set_precomputed_skills_data_ratio()` function that limits the size of the skills data used to raise an error: `PermissionError: [Errno 13] Permission denied:` 

I will investigate if there is another work around for this.

**Comment by leseb** [comment]
> Update:
> 
> Looks like `/usr/share/instructlab/sdg/default_data_recipes/skills.yaml` is no longer editable by the user. This causes the `set_precomputed_skills_data_ratio()` function that limits the size of the skills data used to raise an error: `PermissionError: [Errno 13] Permission denied:`
> 
> I will investigate if there is another work around for this.

https://github.com/opendatahub-io/ilab-on-ocp/pull/230#issuecomment-2517023820

**Comment by MichaelClifford** [comment]
Closed by #230
